src/main/java/net/elytrium/fastmotd/injection/HandshakeSessionHandlerHook.java

```diff
@@ -67,6 +67,10 @@ public boolean handle(LegacyHandshakePacket packet) {
 
   @Override
   public boolean handle(HandshakePacket handshake) {
+    if (handshake.getProtocolVersion() == null || handshake.getProtocolVersion() == ProtocolVersion.UNKNOWN) {
+      handshake.setProtocolVersion(ProtocolVersion.MAXIMUM_VERSION);
```

Comment:
As Velocity does that only while responding to StatusRequestPacket, i think it will be better to move this code here to ensure that nothing got broken:
https://github.com/Elytrium/FastMOTD/blob/a8243a01673f708cc67092c78c54d95e9bb6ff6a/src/main/java/net/elytrium/fastmotd/injection/HandshakeSessionHandlerHook.java#L106-L108

Reply:
Technically, this would affect ping logging by logging replaced version instead of original one.
But we can log before replacing the version, so yes, it will be save to do that.

Reply:
Do you want me to move the current logging so it's before my fix? And/or add logging when the version is replaced?

Reply:
Yes and no.
The current implementation affects not only the status state, but also the login state for no obvious reason.
That's why I want this fix to be moved to the StatusRequestPacket handling, as then it won't affect login state and version logging.